packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -682,160 +903,575 @@ class _SortData {
+  /// result of an order change.
+  static FocusOrder of(BuildContext context, {bool nullOk = false}) {
+    assert(context != null);
+    assert(nullOk != null);
+    final FocusTraversalOrder marker = context.getElementForInheritedWidgetOfExactType<FocusTraversalOrder>()?.widget as FocusTraversalOrder;
+    final FocusOrder order = marker?.order;
+    if (order == null && !nullOk) {
+      throw FlutterError('FocusTraversalOrder.of() was called with a context that '
+          'does not contain a TraversalOrder widget. No TraversalOrder widget '
+          'ancestor could be found starting from the context that was passed to '
+          'FocusTraversalOrder.of().\n'
+          'The context used was:\n'
+          '  $context');
     }
-    return false;
+    return order;
   }
 
+  // Since the order of traversal doesn't affect display of anything, we don't
+  // need to force a rebuild of anything that depends upon it.
   @override
-  bool next(FocusNode currentNode) => _move(currentNode, forward: true);
+  bool updateShouldNotify(InheritedWidget oldWidget) => false;
 
   @override
-  bool previous(FocusNode currentNode) => _move(currentNode, forward: false);
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<FocusOrder>('order', order));
+  }
 }
 
-/// A widget that describes the inherited focus policy for focus traversal.
+/// A widget that describes the inherited focus policy for focus traversal for
+/// its descendants, grouping them into a separate traversal group.
 ///
-/// By default, traverses in widget order using
-/// [ReadingOrderFocusTraversalPolicy].
+/// A traversal group is treated as one entity when sorted by the traversal
+/// algorithm, so it can be used to segregate different parts of the widget tree
+/// that need to be sorted using different algorithms and/or sort orders when
+/// using an [OrderedTraversalPolicy].
+///
+/// By default, traverses in reading order using [ReadingOrderTraversalPolicy].
 ///
 /// See also:
 ///
 ///  * [FocusNode], for a description of the focus system.
-///  * [WidgetOrderFocusTraversalPolicy], a policy that relies on the widget
+///  * [WidgetOrderTraversalPolicy], a policy that relies on the widget
 ///    creation order to describe the order of traversal.
 ///  * [ReadingOrderTraversalPolicy], a policy that describes the order as the
 ///    natural "reading order" for the current [Directionality].
 ///  * [DirectionalFocusTraversalPolicyMixin] a mixin class that implements
 ///    focus traversal in a direction.
+class FocusTraversalGroup extends StatelessWidget {
+  /// Creates a [FocusTraversalGroup] object.
+  ///
+  /// The [child] argument must not be null.
+  FocusTraversalGroup({
+    Key key,
+    FocusTraversalPolicy policy,
+    @required this.child,
+    String debugLabel,
+  })  : _focusNode = FocusNode(
```

Comment:
FocusNode is stateful (can be disposed) so FocusTraversalGroup must be stateful so it can dispose it (since it definitely owns it, since it has a final field and creates it)